lua/entities/gmod_wire_expression2/core/sound.lua

```diff
@@ -303,6 +303,17 @@ local function EmitSound(e2, ent, snd, level, pitch, volume)
         level = maxlevel
     end
 
+    local emitting_sounds = ent:GetVar("E2_emitting_sounds", {})
+    table.insert(emitting_sounds, snd)
+    ent:SetVar("E2_emitting_sounds", emitting_sounds)
+
+    ent:CallOnRemove("E2_EmitSound_stop_all", function()
+        local emitting_sounds = ent:GetVar("E2_emitting_sounds", {})
```

Comment:
Instead of defaulting to an empty table, have a line following `if not emitting_sounds then return end`.